src/common/cockpittest.c

```diff
@@ -481,8 +481,15 @@ _cockpit_assert_data_eq_msg (const char *domain,
     len = strlen (data);
   if (exp_len < 0)
     exp_len = strlen (expect);
-  if (len == exp_len && memcmp (data, expect, len) == 0)
-    return;
+
+  if (len == exp_len)
+    {
+      if (len == 0)
+        return;
+      else if (data && expect && memcmp (data, expect, len) == 0)
+        return;
+    }
+
```

Comment:
Would have been simpler  to just add this to the original condition?

    if (len == 0 || (len == exp_len && memcmp (data, expect, len)) == 0)

But this still looks a bit dubious -- this may just have detected that line 478 is wrong. This *should* say
```C
if (!data || !expect)
    return;
```
not `&&`.